x/oracle/types/oracle.go

```diff
@@ -121,3 +121,10 @@ func (m *Oracle) ValidateOracleCommission(blockTime time.Time, newRate sdk.Dec)
 
 	return nil
 }
+
+func (m *OracleUpgradeInfo) ShouldExecute(ctx sdk.Context) bool {
+	if m.Height > 0 {
+		return m.Height <= ctx.BlockHeight()
```

Comment:
we decision not deleted `uniqueID` for `OracleUpgradeInfo`. 
So I think you need to change condition like below:
```suggestion
		return m.Height == ctx.BlockHeight()
```

Reply:
really good point!

Reply:
good!